### PR TITLE
[MIG] stock_picking_unbuild: Migration to 15.0

### PR DESCRIPTION
**Pull request by nikunjantala**

### CODE REVIEW

**Comment by jans23** [comment]
Please create a PR to OCA instead of this one.

**Comment by jans23** [comment]
Did you do this already?

**Comment by dsolanki-initos** [comment]
@nikunjantala To check functionality of stock_picking_unbuild we need "sale_mrp_link" module in v15.0 so can you please provide me the PR of "sale_mrp_link" in v15.0?

**Comment by jans23** [comment]
There is this PR https://github.com/OCA/sale-workflow/pull/1109
Maybe you have time to read its discussion and suggest how to proceed with sale_mrp_link.

**Review by dsolanki-initos — COMMENTED** [review]
Functional test and code review, In v15.0, do not require this "-*- coding: utf-8 -*-"  so you can remove it.

**Review by ikapasi-initos — CHANGES_REQUESTED** [review]
Code review and function test.
LGTM
Remove line '# -*- coding: utf-8 -*-' in file.

**Comment by nileshsheliya** [comment]
@jans23 Am I need to create PR in https://github.com/OCA/sale-workflow repo, right?

**Comment by jans23** [comment]
Hmm, yes this was my idea. But given the fact that we have many PRs open at OCA we may do so later. Hence, lets leave the module here for now.

**Comment by nileshsheliya** [comment]
Ok.

**Comment by nileshsheliya** [comment]
Code restructure as per OCA standard and also removed unnecessary code.

**Review by manengel — COMMENTED** [review]
I checked the module and tried to test it. Turns out that the dependency sale_mrp_link is not migrated to anything newer than Odoo 12 (or it did move to another repository than sale-workflow?), so there might be a need to investigate what happened to that module and then possibly give that priority over this in terms of migration. EDIT: I already found it out in the meanwhile. But this means that this module must be adjusted to handle the other dependency then?

**Comment by manengel** [comment]
@nikunjantala Any news related to that?

**Comment by dsolanki-initos** [comment]
@manengel @jans23 

I've tried to find PR for the sale_mrp_link in OCA and found that the sale_mrp_link module is available till v12.0 and does not exist from v13.0 in the OCA.

For your reference here is the link of the close PR of sale_mrp_link in v13.0 -> https://github.com/OCA/sale-workflow/pull/1109

> I checked the module and tried to test it. Turns out that the dependency sale_mrp_link is not migrated to anything newer than Odoo 12 (or it did move to another repository than sale-workflow?), so there might be a need to investigate what happened to that module and then possibly give that priority over this in terms of migration. EDIT: I already found it out in the meanwhile. But this means that this module must be adjusted to handle the other dependency then?

**Comment by jans23** [comment]
Please check if sale_mrp_link is still required in 15.0 or perhaps obsolete because of a base feature.

**Comment by dsolanki-initos** [comment]
> Please check if sale_mrp_link is still required in 15.0 or perhaps obsolete because of a base feature.

In the v12.0 functionality of sale_mrp_link, it adds a link between the manufacturing orders and the associated sale order.
A button shown in the form view of sale orders allows the user to access the related manufacturing orders. I have checked in the odoo v15.0 core there is no such functionality.

**Comment by dsolanki-initos** [comment]
sale_mrp_link module is now no more required as this fetaure is available in Odoo base.

File: chatter_confirm_message/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from . import models
```
Comment (manengel) [inline chatter_confirm_message/__init__.py]: Please remove this module from that PR! It is already migrated to Odoo 15.

File: crm_profiling/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from . import models, wizard
```
Comment (manengel) [inline crm_profiling/__init__.py]: Please remove this module from the PR!

File: stock_picking_unbuild/models/stock_picking.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+from odoo import _, api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    is_display_unbuild = fields.Boolean(compute="_compute_display_unbuild", store=True)
```
Comment (manengel) [inline stock_picking_unbuild/models/stock_picking.py:7]: Isn't store=True already the default value or is is_display_unbuild normally created with store=False?

**Review by manengel — CHANGES_REQUESTED** [review]
**Please clean this PR. All the following modules:**
* chatter_confirm_message
* crm_profiling
* email_from
* phone_validation_e164

**have already been migrated in their own PRs.**

File: stock_picking_unbuild/models/stock_picking.py (hunk 3 above)
Comment (dsolanki-initos) [reply stock_picking_unbuild/models/stock_picking.py:7]: > Isn't store=True already the default value or is is_display_unbuild normally created with store=False?

When we use compute attribute of the field so the field does not store the value in the database, we have to add store=True for storing the value.

**Comment by dsolanki-initos** [comment]
> **Please clean this PR. All the following modules:**
> 
> * chatter_confirm_message
> * crm_profiling
> * email_from
> * phone_validation_e164
> 
> **have already been migrated in their own PRs.**

We did not add those modules to this branch, Those commits are visible because we did rebase this branch to the 15.0 branch for up to date and that's why the commits of 15.0 that you mentioned above are visible in this PR.

**Comment by alihashaam** [comment]
@dsolanki-initos 

what is the status here?

**Comment by jans23** [comment]
ready for review

**Comment by alihashaam** [comment]
@jans23 

I have reviewed this PR and it is ready to merge. Just want to mention that the module name is not in auto-installation in odoo.versions.yaml so it needs to be installed manually

**Pull request merged**